### PR TITLE
Famework: Change defaults for Trilinos-specific CMake flags

### LINKED ISSUE #10345: Set Trilinos_MUST_FIND_ALL_TPL_LIBS=TRUE by default for Trilinos 15.0
@trilinos/framework

With Trilinos 14.0 coming up, this would be a good time to change the default for `Trilinos_MUST_FIND_ALL_TPL_LIBS` from `OFF` to `ON`.  (By setting `Trilinos_MUST_FIND_ALL_TPL_LIBS_DEFAULT` to `ON` as described [here](https://tribits.org/doc/TribitsUsersGuide.html#project-name-must-find-all-tpl-libs)).  It was turned off by default because it breaks backward compatibility but it causes problems for new users and new configurations (see, for example, https://github.com/trilinos/Trilinos/issues/10322#issuecomment-1071301714).

So as soon as the Trilinos 13.4 branch is created, we should make this change in the 'develop' branch.

### LINKED ISSUE #12104: Set Trilinos_USE_GNUINSTALLDIRS=ON for Trilinos 15.0
CC: @sebrowne, @ccober6 

## Description

TriBITS has had the ability to use that paths selected by the standard CMake module [GNUInstallDirs.cmake](https://cmake.org/cmake/help/v3.23/module/GNUInstallDirs.html) for a **long** time (see [Setting the install prefix](https://docs.trilinos.org/files/TrilinosBuildReference.html#setting-the-install-prefix)).  But it is turned off in TriBITS by default (for the sake of backward compatibility) and it was never turned on in Trilinos by default (also to maintain backward compatibility).

Therefore, in the goal to move Trilinos and TriBITS to modern CMake ( see TriBITSPub/TriBITS#411), we should hard set the **non-cache** variable `Trilinos_USE_GNUINSTALLDIRS=ON` in the base Trilinos `CMakeLists.txt` file.  This will force every Trilinos user to absorb this change in Trilinos 15.0 (but most customers should never even notice).

Once Trilinos 15.0 comes out, then we can refactor TriBITS to just directly use the standard variables that come from the standard CMake module `include(GNUInstallDirs)`.

Easy as pie (at least for Trilinos and TriBITS).

### PR DESCRIPTION
**Pull request by srbdev**
@trilinos/framework 

## Motivation
See #10345 and #12104

## Related Issues

* Closes #10345 
* Closes #12104

### CODE REVIEW

**Review by sebrowne — DISMISSED** [review]

File: RELEASE_NOTES [hunk 1]
```diff
@@ -1,4 +1,35 @@
...
+###############################################################################
+#                                                                             #
+# Trilinos Release 15.0 Release Notes                                         #
+#                                                                             #
+###############################################################################
+
+CMake
+
+  - Change the default for `Trilinos_MUST_FIND_ALL_TPL_LIBS` from `OFF` to
+    `ON`, by setting `Trilinos_MUST_FIND_ALL_TPL_LIBS_DEFAULT` to `ON`. It was
```
Comment (bartlettroscoe) [inline RELEASE_NOTES:11]:
```suggestion
    `ON`. It was
```

You don't need to tell users how the default was changed.  That is an implementation detail they should not care about.

File: RELEASE_NOTES [hunk 2]
```diff
@@ -1,4 +1,35 @@
...
+    turned off by default because it breaks backward compatibility but it also
+    causes problems for new users and new configurations.
+
+    Users that do not want this new behavior can set
+    `-D Trilinos_MUST_FIND_ALL_TPL_LIBS=OFF`, which is backward compatible.
+
+  - Change the default for `Trilinos_USE_GNUINSTALLDIRS` from `OFF` to `ON`,
+    by setting `Trilinos_USE_GNUINSTALLDIRS_DEFAULT` to `ON`, in the goal to
```
Comment (bartlettroscoe) [inline RELEASE_NOTES:19]:
```suggestion
    in the goal to
```

Again, you don't need to tell the user how the default was changed.  That is an implementation detail.

**Review by bartlettroscoe — CHANGES_REQUESTED** [review]
Just a few small changes.  You can remove some of this detail, IMHO.

**Comment by bartlettroscoe** [comment]
FYI: I thought that I had published this review last week, but I had forgotten to click the final button. Sorry for the delay.

File: RELEASE_NOTES [hunk 3]
```diff
@@ -1,4 +1,35 @@
...
+# Trilinos Release 15.0 Release Notes                                         #
+#                                                                             #
+###############################################################################
+
+CMake
+
+  - Change the default for `Trilinos_MUST_FIND_ALL_TPL_LIBS` from `OFF` to
+    `ON`. It was
+    turned off by default because it breaks backward compatibility but it also
+    causes problems for new users and new configurations.
```
Comment (bartlettroscoe) [inline RELEASE_NOTES:13]: nit: no longer well formatted but no big deal.

File: RELEASE_NOTES [hunk 4]
```diff
@@ -1,4 +1,35 @@
...
+
+    Users that do not want this new behavior can set
+    `-D Trilinos_MUST_FIND_ALL_TPL_LIBS=OFF`, which is backward compatible.
+
+  - Change the default for `Trilinos_USE_GNUINSTALLDIRS` from `OFF` to `ON`,
+    in the goal to
+    move Trilinos and TriBITS to modern CMake.
```
Comment (bartlettroscoe) [inline RELEASE_NOTES:20]: nit: no longer well formatted but no big deal.

**Review by bartlettroscoe — APPROVED** [review]
Looks correct.

nit: formatting could be better since this is a raw file.  In the future, this file should be changed to a *.md file so it will look good when formatted by GitHub and be easier to read.

**Pull request merged**